lib/katello/engine.rb

```diff
@@ -132,9 +132,15 @@ class Engine < ::Rails::Engine
 
     config.to_prepare do
       Katello::CandlepinEventListener.client_factory = proc do
+        settings = SETTINGS[:katello][:candlepin_events].deep_merge(
+          :ssl_key_file => Setting[:ssl_priv_key],
+          :ssl_cert_file => Setting[:ssl_certificate],
+          :ssl_ca_file => Setting[:ssl_ca_file]
```

Comment:
`Setting` requires the database and we already know we had problems with this listener and DB connection slots. Something to watch out for.

Reply:
Should I use `SETTING` here?

Reply:
In practice that will now work because we set them here: https://github.com/theforeman/puppet-foreman/blob/6e867bd927f6aaec12ed4d2140744166589bb962/templates/settings.yaml.erb#L18-L21

But that doesn't cover all deployments and it would tie us to YAML config

Reply:
Btw, I'm not saying it can't be done. Perhaps when it runs there is already a connection anyway. Or wrap in a block:
```ruby
ActiveRecord::Base.connection_pool.with_connection do |connection|
end
```

Reply:
Should I add those to https://github.com/theforeman/foreman/blob/develop/app/services/foreman/env_settings_loader.rb#L29 so we can get them from environment variables or yaml?

Reply:
I actually had some other thought a while back: let's adopt a convention where to store them and rely on defaults.

Reply:
To elaborate on that: I want to introduce some Ruby wrapper that respects `CREDENTIALS_DIRECTORY` (as specified in https://systemd.io/CREDENTIALS/) with some fallback. Then change the implementation of those settings to have defaults and deal with missing files.